iceoryx_utils/cmake/IceoryxPlatformDetection.cmake

```diff
@@ -14,4 +14,16 @@ elseif(WIN32)
     set(ICEORYX_CXX_STANDARD 11)
 elseif(APPLE)
     set(ICEORYX_CXX_STANDARD 17)
-endif(LINUX)
\ No newline at end of file
+endif(LINUX)
+
+set (ICEORYX_WARNINGS -W -Wall -Wextra -Wuninitialized -Wpedantic -Wstrict-aliasing -Wcast-align -Wno-noexcept-type)
```

Comment:
I put this here since the `ICEORYX_CXX_STANDARD` is also here, but this file should probably be rename from `IceoryxPlatformDetection.cmake` to `IceoryxFoo.cmake`. Any suggestions for `Foo`?